lib/rubygems/command.rb

```diff
@@ -152,13 +152,21 @@ def execute
   #--
   # TODO: replace +domain+ with a parameter to suppress suggestions
 
-  def show_lookup_failure(gem_name, version, errors, domain)
+  def show_lookup_failure(gem_name, version, errors, domain, required_by = nil)
+    gem = "'#{gem_name}' (#{version})"
+
     if errors and !errors.empty?
-      msg = "Could not find a valid gem '#{gem_name}' (#{version}), here is why:\n".dup
+      msg = "Could not find a valid gem #{gem}, here is why:\n".dup
       errors.each { |x| msg << "          #{x.wordy}\n" }
       alert_error msg
     else
-      alert_error "Could not find a valid gem '#{gem_name}' (#{version}) in any repository"
+      if required_by && gem != required_by then
+        msg = "Could not find a valid gem #{gem} required by #{required_by} in any repository"
+      else
+        msg = "Could not find a valid gem #{gem} in any repository"
+      end
+
+      alert_error msg
     end
```

Comment:
I think `"Could not find a valid gem #{gem} (required by #{required_by}) in any repository"` may be a bit clearer.

Since both paths end with `alert_error msg`, would something like this work?:

```ruby
    if errors and !errors.empty?
      msg = "Could not find a valid gem #{gem}, here is why:\n".dup
      errors.each { |x| msg << "          #{x.wordy}\n" }
    elsif required_by && gem != required_by then
      msg = "Could not find a valid gem #{gem} (required by #{required_by}) in any repository"
    else
      msg = "Could not find a valid gem #{gem} in any repository"
    end
    alert_error msg
```

Reply:
Awesome tips, thanks!